src/main/java/haste/commands/Parser.java

```diff
@@ -0,0 +1,282 @@
+        } catch (DateTimeParseException e) {
+            return new InvalidCommand("Date Time should be in yyyy-MM-dd HHmm format!");
+        }
+        return new AddCommand(desc, end);
+    }
+
+    /**
+     * Returns a command to add Event.
+     * Verifies user intends to add Event.
+     *
+     * @param cmd User input.
+     * @return AddCommand object containing description, start time and deadline of task.
+     */
+    public static Command verifyEvent(String cmd) {
+        Pattern pt = Pattern.compile("event(( (.*) )?/from( (.*) )?/to( (.*))?)?");
+        Matcher mt = pt.matcher(cmd);
+        mt.find();
+
+        String overall = mt.group(1);
+        String desc = mt.group(3);
+        LocalDateTime start;
+        LocalDateTime end;
+
+        if (Task.checkEmpty(overall) || Task.checkEmpty(desc)) {
+            return new InvalidCommand("Description of Event cannot be empty!");
+        }
+        try {
+            start = Parser.parseTime(mt.group(5));
+            end = Parser.parseTime(mt.group(7));
+        } catch (DateTimeParseException e) {
+            return new InvalidCommand("Date Time should be in yyyy-MM-dd HHmm format!");
+        }
+
+        return new AddCommand(desc, end, start);
+    }
+
+    /**
+     * Returns a LocalDateTime object representing time from user input.
+     *
+     * @param input User input.
+     * @return LocalDateTime object.
+     * @throws DateTimeParseException if user input does not fit accepted time format.
+     */
+    public static LocalDateTime parseTime(String input) throws DateTimeParseException {
+        // formatting command input into LocalDateTime args
+        String formatPattern = "yyyy-MM-dd HHmm";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(formatPattern);
+        LocalDateTime parsedTime = LocalDateTime.parse(input, formatter);
+
+        return parsedTime;
+    }
+
+    /**
+     * Returns String representation of time from LocalDateTime object.
+     *
+     * @param input LocalDateTime object stored in task.
+     * @return String representation of time.
+     */
+    public static String formatTime(LocalDateTime input) {
+        // format time into a string object components
+        String year = String.valueOf(input.getYear());
+        String month = input.getMonth().toString();
+        String day = String.valueOf(input.getDayOfMonth());
+        int hourAndTime = input.getHour() * 100 + input.getMinute();
+        // ensure that hourAndTime would always have 4 digits
+        return hourAndTime < 1000
+                ? month + " " + day + " " + year + " 0" + (hourAndTime)
+                : month + " " + day + " " + year + " " + (hourAndTime);
+    }
+
+    /**
+     * Returns String representation of LocalDateTime object in user input format.
+     *
+     * @param savedTime LocalDateTime object stored in task.
+     * @return String representation of time.
+     */
+    public static String getCmd(LocalDateTime savedTime) {
```

Comment:
Perhaps you can consider getStringFormat or getCmdFormat as a method name for better clarity?

Reply:
Thanks for the suggestion! will rectify